packages/core/test-e2e/insured-bridge/helpers/MultiProviderHelper.js

```diff
@@ -0,0 +1,67 @@
+const hre = require("hardhat");
+
+const fs = require("fs");
+const path = require("path");
+
+function getProviders() {
+  return {
+    l1RpcProvider: new hre.ethers.providers.JsonRpcProvider("http://localhost:9545"),
+    l2RpcProvider: new hre.ethers.providers.JsonRpcProvider("http://localhost:8545"),
+  };
+}
+
+function getAllFiles(dirPath, arrayOfFiles = []) {
+  const files = fs.readdirSync(dirPath);
+
+  arrayOfFiles = arrayOfFiles || [];
+
+  files.forEach((file) => {
+    if (fs.statSync(dirPath + "/" + file).isDirectory()) arrayOfFiles = getAllFiles(dirPath + "/" + file, arrayOfFiles);
+    else arrayOfFiles.push(path.join(dirPath, "/", file));
+  });
+
+  return arrayOfFiles;
+}
+
+function findArtifactFromPaths(contractName, artifactsPath) {
+  const allArtifactsPaths = getAllFiles(artifactsPath);
+  const desiredArtifactPaths = allArtifactsPaths.filter((a) => a.endsWith(`/${contractName}.json`));
+
+  if (desiredArtifactPaths.length !== 1)
+    throw new Error(`Couldn't find desired artifact or found too many for ${contractName}`);
+  return JSON.parse(fs.readFileSync(desiredArtifactPaths[0], "utf-8"));
+}
+
+function getOptimismArtifact(contractName, ovm = false) {
```

Comment:
So this returns all artifacts that come with the eth-optimism npm package? This is kind of ugly and will be a pain to maintain as relative paths move around but I can't see a better way unless `eth-optimism` exports a helper

Reply:
Actually, why not add the required eth-optimism contracts to our `core/contracts/external` directory?

Reply:
it just returns the one artifact of the particular `contractName` you provided., not all artifacts. it will search through all `artifacts` the package has though.

I agree it's not ideal. ideally, we should use the `getContractFactory` that the optimism `@eth-optimism/contracts` package exports. However, this does not work consistently for some reason and I have a bunch of problems actually using this function in these tests. Hopefully they fix this in the future (or add other ways of getting these artefacts) and then we can remove this logic and rather use their methods for this kind of call.

Reply:
Regarding putting it in `core/contracts/external`, that is possible but IMHO it's a bit messy. we don't need those artifacts anywhere in the repo and it's actually a fair amount of code that we'd have to copy over.